File: backend/internal/model/calendar.go
package model

type CalendarProjectSource struct {
	ProjectID  string `json:"project_id"`
	Name       string `json:"name"`
	Type       string `json:"type"`
	Enabled    bool   `json:"enabled"`
	Default    bool   `json:"default"`
	Color      string `json:"color"`
	OrderIndex int    `json:"order_index"`
}

type CalendarProjectSourcesResponse struct {
	Sources           []CalendarProjectSource `json:"sources"`
	AvailableProjects []CalendarProjectSource `json:"available_projects"`
}

type CalendarProjectSourceInput struct {
	ProjectID  string `json:"project_id"`
	Enabled    bool   `json:"enabled"`
	Color      string `json:"color"`
	OrderIndex int    `json:"order_index"`
}

type SaveCalendarProjectSourcesRequest struct {
	Sources []CalendarProjectSourceInput `json:"sources"`
}

File: frontend/src/api/calendar.ts
import { api } from './client'

export interface CalendarProjectSource {
  project_id: string
  name: string
  type: string
  enabled: boolean
  default: boolean
  color: string
  order_index: number
}

export interface CalendarProjectSourcesResponse {
  sources: CalendarProjectSource[]
  available_projects: CalendarProjectSource[]
}

export interface SaveCalendarProjectSourceInput {
  project_id: string
  enabled: boolean
  color: string
  order_index: number
}

export interface SaveCalendarProjectSourcesRequest {
  sources: SaveCalendarProjectSourceInput[]
}

export async function getCalendarProjectSources() {
  const res = await api.get<CalendarProjectSourcesResponse>(
    '/api/calendar/project-sources',
  )
  return res.data
}

export async function saveCalendarProjectSources(
  body: SaveCalendarProjectSourcesRequest,
) {
  const res = await api.put<CalendarProjectSourcesResponse>(
    '/api/calendar/project-sources',
    body,
  )
  return res.data
}
